governator-core/src/main/java/com/netflix/governator/internal/PostConstructLifecycleActions.java

```diff
@@ -16,39 +19,59 @@
  * 
  * @author elandau
  */
-public final class PostConstructLifecycleActions extends AbstractLifecycleFeature {
+public final class PostConstructLifecycleActions extends AbstractLifecycleFeature<Set<String>> {
 
     public static PostConstructLifecycleActions INSTANCE = new PostConstructLifecycleActions();
     
     private PostConstructLifecycleActions() {
-        
+        super(IntrospectionFlag.METHOD, IntrospectionFlag.SUPERCLASS);
     }
     
     @Override
-    protected List<LifecycleAction> getMethodActions(final Class<?> type, final Method method) {
-        if (null != method.getAnnotation(PostConstruct.class)) {
-            method.setAccessible(true);
-            return Collections.<LifecycleAction>singletonList(new LifecycleAction() {
-                @Override
-                public void call(Object obj) throws IllegalAccessException, IllegalArgumentException, InvocationTargetException {
-                    method.invoke(obj);
-                }
-                
-                @Override
-                public String toString() {
-                    return new StringBuilder()
-                        .append("PostConstruct[")
-                        .append(type.getName())
-                        .append("#")
-                        .append(method.getName())
-                        .append("]")
-                        .toString();
-                }
-            });
-        }
+    protected List<LifecycleAction> getMethodActions(Set<String> typeContext, final Class<?> type, final Method method) {
+    	int modifiers = method.getModifiers();
+		if (!Modifier.isStatic(modifiers) && !Modifier.isAbstract(modifiers) && method.getParameterCount() == 0  && Void.TYPE.equals(method.getReturnType())) {
+			if (!typeContext.contains(method.getName())) {
+		        if (null != method.getAnnotation(PostConstruct.class)) {
+			        if (!method.isAccessible()) {
```

Comment:
no need to make a Method accessible if it already is; I'd expect .isAccessible() is cheaper call than .setAccesible() even in the trivial case.